### DISCUSSION TITLE
Dynamic URLs with ISR

### DISCUSSION THREAD

**Discussion by fxmb** [iamvishnusankar/next-sitemap#601] [Q&A]
We have around 100k dynamic pages all under the same directory. We do not build them at `yarn build` but serve them via ISR on demand. The arising problem is that with next-sitemap they are not being added to the sitemap anymore - since they are not build. How would I best run a snippet that fetches all slugs from the server, and adds these 100k pages manually to the sitemap.xml without having to actually build them? 

Thanks for any pointers or help :-)

**Comment by iamvishnusankar** [comment]
@fxmb You may use [Dynamic Sitemaps](https://github.com/iamvishnusankar/next-sitemap#generating-dynamicserver-side-sitemaps)

**Reply by fxmb** [reply]
Thanks so much for the fast and precise answer.

So I would basically just implement this snippet?

```
// app/server-sitemap-index.xml/route.ts
import { getServerSideSitemapIndex } from 'next-sitemap'

export async function GET(request: Request) {
  // Method to source urls from cms
  // const urls = await fetch('https//example.com/api')

  return getServerSideSitemapIndex([
    'https://example.com/path-1.xml',
    'https://example.com/path-2.xml',
  ])
}
```

**Reply by fxmb** [reply]
@iamvishnusankar the signature of `getServerSideSitemapIndex` is different: it expects a context object as a first argument. Am I missing something here or are the docs wrong?
